gradio/blocks.py

```diff
@@ -320,6 +322,11 @@ def render(self):
         if Context.block is not None:
             Context.block.children.extend(self.children)
 
+    async def create_limiter(self, max_threads: Optional[int]):
```

Comment:
Hmm, why is this a async function?

Reply:
I got an error when I tried to do it inline in `launch`, or using a regular function. `CapacityLimiter` refuses to run outside an async context, I think it attempts to retrieve the async library currently in use, and fails:

```
Exception has occurred: AsyncLibraryNotFoundError
unknown async library, or not in async context
```

Reply:
Ok then I will resolve conflicts 👍